### PR TITLE
[BUGFIX] Always delete tmp handlers when called regardless of return value

### PR DESCRIPTION
**Pull request by christophersansone**
I noticed a problem with #register_tmp_handler sometimes firing the same handler multiple times.  The issue had to do with the return value: if falsy, it would re-execute.  This also made it difficult to throw :pass, because it would then only remove the handler if subsequent handler was registered.

Also, presumably, the handler may have been deleted even if it was not actually called.

I simply changed the logic to delete the handler if it was marked as temporary and if it had been called, regardless of its return value.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,12 +78,12 @@ a.register_handler_with_options(:event, {:tmp => true, :priority => 10}, :foo =>
...
 
-When multiple handlers match the event, the return value of each handler will determine if the handler chain continues. A truthy return value will cause the handler to swallow the event and halt the handler chain. A falsy return value will continue the chain.
+Each handler can control whether subsequent handlers should be executed by throwing `:pass` or `:halt`.
 
-It is possible to explicitly pass to the next handler by throwing `:pass` in your handler:
+To explicitly pass to the next handler, throw `:pass` in your handler:
 
 ```ruby
-a.register_handler(:event) { throw :pass }
+a.register_handler(:event) { do\_stuff; throw :pass }
```
Comment (benlangfeld) [inline README.md]: Why the backslashes here?

**Comment by benlangfeld** [comment]
Wow, I have no idea how I missed this. @victorluft just pointed it out to me. Sorry!

The change looks good, except for the fact that it removes documented functionality (though it seems un-tested, hmm), which I am concerned that people might be relying on. We'll take a closer look at this.

**Comment by benlangfeld** [comment]
Ok, I tell a lie, this behaviour is not currently in, as asserted by this: https://github.com/adhearsion/has-guarded-handlers/commit/300180bfdebe3097add0bda83b1478d57ac353e2. I guess at some point in the past I already dealt with that and forgot to remove it from the README. I'm going to go ahead and merge this shortly, thanks!

**Comment by christophersansone** [comment]
Haha, I was just thinking about this last night.  Awesome, thanks for taking a look!

As for the README, the underscores need to be escaped by backslashes because Markdown uses underscores to indicate emphasis (italics).

**Comment by benlangfeld** [comment]
> As for the README, the underscores need to be escaped by backslashes because Markdown uses underscores to indicate emphasis (italics).

Even in code fences? If you look at the rendered version, the backslash isn't escaping, it actually shows up.

**Comment by benlangfeld** [comment]
If you've got a moment, a changelog entry and commit squash would mean I can just hit the merge button straight away ;)

**Comment by christophersansone** [comment]
Hmm, yeah you're right about the backslashes, at least how Github's Markdown parser handles it.  I probably previewed it with a different parser that handled it differently.

I'll add the changelog entry and squash the commits.

**Comment by christophersansone** [comment]
Okay, should be good to go.  Thanks Ben!

**Comment by christophersansone** [comment]
Shoot sorry, I must have blown the squash.  Let me try again...

**Comment by christophersansone** [comment]
Okay, the specs were working fine for me until I did a fresh `bundle install`.  It pulled down rspec 3.2, when the specs were written as rspec 2.  The gemspec is requiring `rspec >= 2.5.0`.  I can fix this by either locking rspec into 2.5 or updating the specs.  @benlangfeld do you have a preference?

**Comment by benlangfeld** [comment]
I'd prefer to upgrade to RSpec 3, but realise that's more demanding. If you have time, awesome! If not, lets just lock to 2.

**Comment by christophersansone** [comment]
Happy to upgrade to RSpec 3... it's the least I can do to contribute to Adhearsion.  It's done and all tests pass, so we are good to go.

**Comment by benlangfeld** [comment]
Released as 1.6.2. Thanks!
